### PR TITLE
Made loading wide files into dialogs more efficient

### PR DESCRIPTION
**Pull request by Patowhiz**
Fixes partly #7162.
@africanmathsinitiative/developers this is ready for review.

### CODE REVIEW

**Comment by Patowhiz** [comment]
When testing this I noticed there is a bug when importing spss data, for instance the spss dataset in issue #7161.
I'm not sure what's causing the issue. So @rdstern you may test the loading of dialogs using the `poulet` dataset from package `factoMineR`

**Review by rdstern — DISMISSED** [review]
@Patowhiz and @shadrackkibet I tried with the poulet data as you suggested.  The selector works fine now.  I have not checked back to the time difference, but this is very satisfactory now.  So I am approving.
I hope @shadrackkibet can check the code and merge, when he is happy.  Perhaps, at the same time he could try the steps I took, because I seemed to find problems with his select.  This should not affect the merging of this, but I wonder what is happening with select?
a) I wanted to see whether the selector would cope well when there was a selection.  So I selected all variables that start with A.  It took a long time to make the selection - then it turned the names red - as desired, but did not seem to make the select that I thought.
b) Then I tried the row names dialogue - and this gives an error with selected data - I must check if that is general, and will report.
However these do not affect that I think this is now working for these very wide datasets.  I hope this can be merged and look forward to the other efficiency moves - loading the data, and also factor variables with many levels.  But this is a great first step.

**Comment by rdstern** [comment]
@lloyddewit I was happy to approve this well over a week ago.  @shadrackkibet has been quiet, so I wonder if you could check and merge if ok.  My comments above were in relation to Shadrack's select work, rather than this pull request.  If we merge then I'll copy them into a new issue for select.

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@Patowhiz Thank you for the clear and well-commented code. Just a couple of small questions

File: instat/ucrSelector.vb [hunk 1]
```diff
@@ -49,6 +49,12 @@ Public Class ucrSelector
     '''</summary>
     Private lstOrderedReceivers As New List(Of ucrReceiver)
 
+    ''' <summary>
+    ''' stores the list selector's list view 'fill conditions'
+    ''' used to check if there is need to clear and refill list view based on supplied parameters
+    ''' </summary>
+    Private _strCurrentSelectorFillCondition As String = ""
```
[the remark this answers is not in the document]
Comment (Patowhiz) [reply instat/ucrSelector.vb:56]: @lloyddewit  I thought of that too but I had an issue with code clarity though that's arguable (the scope and initialization aspect of static variable inside functions and how fundamentally different they are in VB.Net compared to other OOP languages.)
I have no strong objection about it (it's a VB.NET feature anyway )so I have just replaced the global variable with the static function scoped variable as you have suggested.

**Comment by Patowhiz** [comment]
@lloyddewit ready for your review.

**Review by lloyddewit — APPROVED** [review]
@N-thony Thank you! happy to approve

**Comment by lloyddewit** [comment]
@rdstern Only a small change since your last approval, please could you test/approve?
Thanks

**Pull request merged**